pkg/apis/awsprovider/v1alpha1/register.go

```diff
@@ -81,3 +81,37 @@ func MachineStatusFromProviderStatus(extension *runtime.RawExtension) (*AWSMachi
 
 	return status, nil
 }
+
+func EncodeMachineStatus(status *AWSMachineProviderStatus) (*runtime.RawExtension, error) {
+	if status == nil {
+		return &runtime.RawExtension{}, nil
+	}
+
+	var rawBytes []byte
+	var err error
+	if rawBytes, err = yaml.Marshal(status); err != nil {
+		return nil, err
+	}
+
+	return &runtime.RawExtension{
+		Raw: rawBytes,
+		//Object: ??,
+	}, nil
+}
+
+func EncodeClusterStatus(status *AWSClusterProviderStatus) (*runtime.RawExtension, error) {
+	if status == nil {
+		return &runtime.RawExtension{}, nil
+	}
+	var rawBytes []byte
+	var err error
+
+	if rawBytes, err = yaml.Marshal(status); err != nil {
+		return nil, err
+	}
```

Comment:
The above could be simplified a little
```
rawBytes, err := yaml.Marshal(status)
if err != nil {
		return nil, err
	}
```
same for the `EncodeMachineStatus`

Reply:
This will goaway when we switch to the api machinery conversion.
I will do that as a part of this PR

Reply:
Looks like we'll need to revisit the api-machinery conversion. Will open an issue for that if this remains unsolved.

Reply:
Created https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/306